llama-cpp-sys-2/build.rs

```diff
@@ -1,1181 +1,346 @@
-            // cx.flag("-mavx512vnni").flag("-mfp16-format=ieee");
-            // cxx.flag("-mavx512vnni").flag("-mfp16-format=ieee");
-        } else if cfg!(target_family = "windows") {
-            cx.define("__ARM_NEON", None)
-                .define("__ARM_FEATURE_FMA", None)
-                .define("__ARM_FEATURE_DOTPROD", None)
-                .define("__aarch64__", None);
-            cxx.define("__ARM_NEON", None)
-                .define("__ARM_FEATURE_FMA", None)
-                .define("__ARM_FEATURE_DOTPROD", None)
-                .define("__aarch64__", None);
+fn get_cargo_target_dir() -> Result<std::path::PathBuf, Box<dyn std::error::Error>> {
+    let out_dir = std::path::PathBuf::from(std::env::var("OUT_DIR")?);
+    let profile = std::env::var("PROFILE")?;
+    let mut target_dir = None;
+    let mut sub_path = out_dir.as_path();
+    while let Some(parent) = sub_path.parent() {
+        if parent.ends_with(&profile) {
+            target_dir = Some(parent);
+            break;
         }
+        sub_path = parent;
     }
+    let target_dir = target_dir.ok_or("not found")?;
+    Ok(target_dir.to_path_buf())
 }
 
-/// Add platform appropriate flags and definitions for compilation warnings.
-fn push_warn_flags(cx: &mut Build, cxx: &mut Build) {
-    if cfg!(target_family = "unix") {
-        cx.flag("-pthread")
-            .flag("-Wall")
-            .flag("-Wextra")
-            .flag("-Wpedantic")
-            .flag("-Wcast-qual")
-            .flag("-Wdouble-promotion")
-            .flag("-Wshadow")
-            .flag("-Wstrict-prototypes")
-            .flag("-Wpointer-arith");
-        cxx.flag("-fPIC")
-            .flag("-pthread")
-            .flag("-Wall")
-            .flag("-Wdeprecated-declarations")
-            .flag("-Wextra")
-            .flag("-Wpedantic")
-            .flag("-Wcast-qual")
-            .flag("-Wno-unused-function")
-            .flag("-Wno-multichar");
-    } else if cfg!(target_family = "windows") {
-        cx.flag("/W4")
-            .flag("/Wall")
-            .flag("/wd4820")
-            .flag("/wd4710")
-            .flag("/wd4711")
-            .flag("/wd4820")
-            .flag("/wd4514");
-        cxx.flag("/W4")
-            .flag("/Wall")
-            .flag("/wd4820")
-            .flag("/wd4710")
-            .flag("/wd4711")
-            .flag("/wd4820")
-            .flag("/wd4514");
+fn copy_folder(src: &Path, dst: &Path) {
+    std::fs::create_dir_all(dst).expect("Failed to create dst directory");
+    if cfg!(unix) {
+        std::process::Command::new("cp")
+            .arg("-rf")
+            .arg(src)
+            .arg(dst.parent().unwrap())
+            .status()
+            .expect("Failed to execute cp command");
+    }
+
+    if cfg!(windows) {
+        std::process::Command::new("robocopy.exe")
+            .arg("/e")
+            .arg(src)
+            .arg(dst)
+            .status()
+            .expect("Failed to execute robocopy command");
     }
 }
```

Comment:
this should probably fail if neither `cfg!` blocks run. (if unix / else if windows / else fail)